src/hotspot/share/opto/superword.hpp

```diff
@@ -209,56 +207,89 @@ class SuperWord : public ResourceObj {
 
   GrowableArray<Node_List*> _packset;    // Packs for the current block
 
-  GrowableArray<int> &_bb_idx;           // Map from Node _idx to index within block
-
-  GrowableArray<Node*> _block;           // Nodes in current block
-  GrowableArray<PhiNode*> _mem_slice_head; // Memory slice head nodes
-  GrowableArray<MemNode*> _mem_slice_tail; // Memory slice tail nodes
   GrowableArray<SWNodeInfo> _node_info;  // Info needed per node
   CloneMap&            _clone_map;       // map of nodes created in cloning
   MemNode const* _align_to_ref;          // Memory reference that pre-loop will align to
 
   DepGraph _dg; // Dependence graph
 
-  // Scratch pads
-  GrowableArray<Node*> _nlist; // List of nodes
-
  public:
-  SuperWord(const VLoop &vloop, VSharedData &vshared);
+  SuperWord(const VLoopAnalyzer &vloop_analyzer);
 
   // Attempt to run the SuperWord algorithm on the loop. Return true if we succeed.
   bool transform_loop();
 
   // Decide if loop can eventually be vectorized, and what unrolling factor is required.
   static void unrolling_analysis(const VLoop &vloop, int &local_loop_unroll_factor);
 
+  // VLoopAnalyzer Accessors
+  const VLoopAnalyzer& vloop_analyzer() const { return _vloop_analyzer; }
+
   // VLoop Accessors
-  const VLoop& vloop()        const { return _vloop; }
-  PhaseIdealLoop* phase()     const { return vloop().phase(); }
-  PhaseIterGVN& igvn()        const { return vloop().phase()->igvn(); }
-  IdealLoopTree* lpt()        const { return vloop().lpt(); }
-  CountedLoopNode* cl()       const { return vloop().cl(); }
-  PhiNode* iv()               const { return vloop().iv(); }
-  int iv_stride()             const { return cl()->stride_con(); }
-  bool in_bb(const Node* n)   const { return vloop().in_bb(n); }
+  const VLoop& vloop()                  const { return vloop_analyzer().vloop(); }
+  PhaseIdealLoop* phase()               const { return vloop().phase(); }
+  PhaseIterGVN& igvn()                  const { return vloop().phase()->igvn(); }
+  IdealLoopTree* lpt()                  const { return vloop().lpt(); }
+  CountedLoopNode* cl()                 const { return vloop().cl(); }
+  PhiNode* iv()                         const { return vloop().iv(); }
+  int iv_stride()                       const { return cl()->stride_con(); }
```

Comment:
Could these also be field? You access these quite a lot but only inside this class and they seem to be constant.

Reply:
I don't think it matters for performance. I was more thinking that it's easier to have immutable fields over const methods - but I guess it's also a matter of taste.